extensions/interactions/MathExpressionInput/MathExpressionInput.js

```diff
@@ -30,13 +30,127 @@ oppia.directive('oppiaInteractiveMathExpressionInput', [
+           * Adds a button overlay and invisible text field used to bring up
+           * the keyboard on mobile devices.
+           *
+           * TODO(Oishikatta): On iOS/Safari, keyboard may only appear on first
+           * press. This may not be a significant issue as the
+           * MathExpressionInput is recreated if the given answer is incorrect.
+           */
+          var makeGuppyMobileFriendly = function() {
+            /**
+             * Checks if the guppy div has a width and height greater than 0,
+             * if not schedules a timeout to run again after 100ms. If the
+             * guppy div has a valid width/height, position the invisible
+             * button directly over the guppy div. For mobile browsers, focus()
+             * can only be called from within an onclick handler. Using a form
+             * element was more reliable than attaching the handler to the
+             * guppy div directly.
+             */
+            var positionButtonOverlay = function() {
+              var guppyOffset = $(guppyDivElt).position();
+              var guppySize = guppyDivElt.getBoundingClientRect();
+
+              // If the guppy div hasn't rendered yet, retry after 100ms.
+              if (guppySize.width === 0 || guppySize.height === 0) {
+                $timeout(positionButtonOverlay, 100);
+              } else {
+                $('#startMathInputButton').css({
+                  top: guppyOffset.top,
+                  left: guppyOffset.left,
+                  width: guppySize.width,
+                  height: guppySize.height
+                });
+              }
+            };
+            positionButtonOverlay();
+
+            // The focus() call must be in a click event handler and on a text
+            // field to make the mobile keyboard appear.
+            $scope.startMobileMathInput = function() {
+              guppyInstance.activate();
+
+              var fakeInputElement = document.querySelector(
+                '#fakeInputForMathExpression');
+              fakeInputElement.focus();
+
+              // Place the cursor at the end of the text input, so that the
+              // user can use backspace to delete.
+              fakeInputElement.setSelectionRange(
+                fakeInputElement.value.length, fakeInputElement.value.length);
+            };
+
+            // Mapping characters back to Mousetrap codes.
+            // Use Mousetrap.trigger(code) to replay them.
+            var K_SYM_REVERSE_MAP = {
+              '^': 'shift+6',
+              '*': 'shift+8',
+              '(': 'shift+9',
+              '<': 'shift+,',
+              '>': 'shift+.',
+              '\\': 'shift+\\',
+              ' ': 'space',
+              ')': 'shift+0'
+              // Missing up/down. user needs to use [space], 'sub' instead.
+              // Mobile devices typically don't have arrow keys anyways.
+            };
+
+            var setGuppyContentFromInput = function() {
```

Comment:
Did you check what happens if the user hits the Backspace key -- is everything still OK? (I guess so, but just thought it worth checking.)